### PR TITLE
storage: Support "http" method of clevis 

### PR DESCRIPTION
**Pull request by mvollmer**
- [x] #9005 

Video: https://www.youtube.com/watch?v=68Qu_vpoKAQ

### CODE REVIEW

File: pkg/storaged/clevis-dialogs.jsx [hunk 1]
```diff
@@ -20,40 +20,84 @@
...
     dialog_open({ Title: _("Add network key"),
                   Fields: [
-                      TextInput("url", _("Key server address"),
-                                { validate: val => !val.length && _("Server address cannot be empty") }),
+                      SelectOne("method", _("Method"), { },
+                                [ { value: "tang",
+                                    title: _("\"tang\" Binding server") },
+                                  { value: "http",
+                                    title: _("\"http\" Key escrow") }
+                                ]),
```
Comment (martinpitt) [inline pkg/storaged/clevis-dialogs.jsx]: Should this get some explanatory popup help? These are fairly technical slang, and not self-explanatory what the difference is.

File: pkg/storaged/clevis-dialogs.jsx [hunk 2]
```diff
@@ -20,40 +20,84 @@
...
+                                  },
+                                  visible: vals => vals.method == "http"
+                                }),
+                      CheckBox("allow_plain_http", _("Allow \"http://\" URL"),
+                               { visible: vals => vals.method == "http",
+                                 validate: (val, vals) => {
+                                     if (vals.http_url.startsWith("http:") && !val)
+                                         return _("This box must be checked to confirm that the key will be transported without HTTPS");
+                                 }
+                               }),
```
Comment (martinpitt) [inline pkg/storaged/clevis-dialogs.jsx]: This feels rather strange to me..

File: pkg/storaged/clevis-dialogs.jsx [hunk 3]
```diff
@@ -20,40 +20,84 @@
...
+                      SelectOne("http_method", _("HTTP method"),
+                                { visible: vals => vals.method == "http" },
+                                [ { value: "PUT", title: "PUT" },
+                                  { value: "POST", title: "POST" }
+                                ]),
```
Comment (martinpitt) [inline pkg/storaged/clevis-dialogs.jsx]: Are there different tang versions out in the wild which only support one but not the other? There is no way the user can know what is right here; could it just try both?

File: pkg/storaged/clevis-dialogs.jsx (hunk 2 above)
Comment (mvollmer) [reply pkg/storaged/clevis-dialogs.jsx]: It is strange.  I copied the "clevis http encrypt" configuration options pretty verbatim into the dialog.  The way we could do it is to just have a checkbox: "[ ] Use TLS" (on by default) and then adjust the URL accordingly.

File: pkg/storaged/clevis-dialogs.jsx (hunk 3 above)
Comment (mvollmer) [reply pkg/storaged/clevis-dialogs.jsx]: Tang is not involved here at all.  I assume there are different kinds of escrow servers out there which might want to be talked to differently.  For the tests, I made a mock server in Python.

If there are some in Fedora / RHEL, maybe as part of FreeIPA, we can have presets for those, I guess.

File: pkg/storaged/clevis-dialogs.jsx (hunk 1 above)
Comment (mvollmer) [reply pkg/storaged/clevis-dialogs.jsx]: Yeah.  We need to seriously bikeshed the terminology.  I kinda made it awful on purpose for the playground.

File: pkg/storaged/clevis-dialogs.jsx [hunk 4]
```diff
@@ -20,40 +20,84 @@
...
+                      SelectOne("key_type", _("Type"),
+                                { visible: vals => vals.method == "http" },
+                                [ { value: "octet-stream", title: "octect-stream" },
```
Comment (martinpitt) [inline pkg/storaged/clevis-dialogs.jsx]: "octect" typo

Comment (mvollmer) [reply pkg/storaged/clevis-dialogs.jsx]: Fixed

**Comment by martinpitt** [comment]
Adding back needs-rebase. Please rebase against latest master to ensure that this still works with current eslint rules.

**Comment by martinpitt** [comment]
Nevermind, you already did. Do you consider this good enough for landing now, or do you plan to do more work on it?

**Comment by mvollmer** [comment]
> Do you consider this good enough for landing now, or do you plan to do more work on it?

It's good enough for me.

**Pull request closed without merging**